mParticle-Apple-SDK/MPRokt.m

```diff
@@ -0,0 +1,132 @@
+              attributes:(NSDictionary<NSString *, NSString *> * _Nullable)attributes
+              placements:(NSDictionary<NSString *, MPRoktEmbeddedView *> * _Nullable)placements
+               callbacks:(MPRoktEventCallback *)callbacks {
+    NSArray<NSDictionary<NSString *, NSString *> *> *attributeMap = [self getRoktPlacementAttributesMapping];
+    
+    // If attributeMap is nil the kit hasn't been initialized
+    if (attributeMap) {
+        NSMutableDictionary *mappedAttributes = attributes.mutableCopy;
+        for (NSDictionary<NSString *, NSString *> *map in attributeMap) {
+            NSString *mapFrom = map[@"map"];
+            NSString *mapTo = map[@"value"];
+            if (mappedAttributes[mapFrom]) {
+                NSString * value = mappedAttributes[mapFrom];
+                [mappedAttributes removeObjectForKey:mapFrom];
+                mappedAttributes[mapTo] = value;
+            }
+        }
+        for (NSString *key in mappedAttributes) {
+            [[MParticle sharedInstance].identity.currentUser setUserAttribute:key value:mappedAttributes[key]];
+        }
+        
+        dispatch_async(dispatch_get_main_queue(), ^{
+            // Forwarding call to kits
+            MPForwardQueueParameters *queueParameters = [[MPForwardQueueParameters alloc] init];
+            [queueParameters addParameter:identifier];
+            [queueParameters addParameter:[self confirmSandboxAttribute:mappedAttributes]];
+            [queueParameters addParameter:placements];
+            [queueParameters addParameter:callbacks];
+            
+            SEL roktSelector = @selector(executeWithViewName:attributes:placements:callbacks:filteredUser:);
+            [[MParticle sharedInstance].kitContainer_PRIVATE forwardSDKCall:roktSelector
+                                                                      event:nil
+                                                                 parameters:queueParameters
+                                                                messageType:MPMessageTypeEvent
+                                                                   userInfo:nil
+            ];
+        });
+    } else {
+        MPILogVerbose(@"[MParticle.Rokt selectPlacements: not performed since Kit not configured");
+    }
+}
+
+- (NSArray<NSDictionary<NSString *, NSString *> *> *)getRoktPlacementAttributesMapping {
+    NSArray<NSDictionary<NSString *, NSString *> *> *attributeMap = nil;
+    
+    // Get the kit configuration
+    NSArray<NSDictionary *> *kitConfigs = [MParticle sharedInstance].kitContainer_PRIVATE.originalConfig.copy;
+    NSDictionary *roktKitConfig;
+    for (NSDictionary *kitConfig in kitConfigs) {
+        if (kitConfig[@"id"] != nil && [kitConfig[@"id"] integerValue] == 181) {
+            roktKitConfig = kitConfig;
+        }
+    }
+    
+    // Get the placement attributes map
+    NSString *strAttributeMap;
+    NSData *dataAttributeMap;
+    if (roktKitConfig != nil) {
```

Comment:
If you check for `!roktKitConfig` above, you can remove this conditional here and make it a little cleaner.